js/src/admin/components/AuthPage.js

```diff
@@ -0,0 +1,215 @@
+            <div className="helpText">
+              {app.translator.trans('core.admin.auth.text')}
+            </div>
+
+            <fieldset class="AuthPage-settings">
+              {Switch.component({
+                state: this.values.enable_password_auth(),
+                onchange: this.values.enable_password_auth,
+                disabled: this.allSsoDisabled(),
+                children: app.translator.trans('core.admin.auth.enable_password_auth_label'),
+              })}
+              {Switch.component({
+                state: this.values.allow_sign_up(),
+                onchange: this.values.allow_sign_up,
+                children: app.translator.trans('core.admin.auth.allow_sign_up_label'),
+              })}
+            </fieldset>
+
+            <fieldset class="AuthPage-sso">
+              <legend>{app.translator.trans('core.admin.auth.sso.heading')}</legend>
+              <div className="helpText">
+                {app.translator.trans('core.admin.auth.sso.help_text')}
+              </div>
+              {Object.keys(this.drivers).length > 0 ?
+                <table className="SsoGrid">
+                  <thead>
+                    <tr>
+                      <td />
+                      {this.driverFields.map(field => (
+                        <th className="SsoGrid-groupToggle">
+                          {icon(field.icon)} {field.label}
+                        </th>
+                      ))}
+                    </tr>
+                  </thead>
+
+                  <tbody>
+                    {Object.keys(this.drivers).map(driver => (
+                      <tr>
+                        <td className="SsoGrid-groupToggle">
+                          {icon(this.drivers[driver].icon)} {this.drivers[driver].name || driver}
+                        </td>
+                        {this.driverFields.map(field => (
+                          <td className="SsoGrid-checkbox">
+                            {this.driverInputs[this.driverFieldKey(field.name, driver)]}
+                          </td>
+                        ))}
+                      </tr>
+                    ))}
+                  </tbody>
+                </table>
+                : <div className="helpText">
+                  {app.translator.trans('core.admin.auth.no_sso_drivers_found')}
+                </div>}
+            </fieldset>
+
+            {Button.component({
+              type: 'submit',
+              className: 'Button Button--primary',
+              children: app.translator.trans('core.admin.auth.submit_button'),
+              disabled: !this.changed()
+            })}
+          </form>
+        </div>
+      </div>
+    );
+  }
+
+  toggle(key) {
+    this.values[key](!this.values[key]());
+
+    this.driverInputs[key].props.state = this.values[key]();
+
+    if (this.allSsoDisabled()) {
+      this.values['enable_password_auth'](true);
+      m.redraw();
```

Comment:
This `m.redraw()` is unnecessary as there's another just below outside of the `if`

Reply:
Fixed